cleaner/environment_matrix/classes/clean.php

```diff
@@ -96,7 +101,34 @@ static public function execute() {
                         }
 
                         if (!$dryrun) {
-                            set_config($config->config, $config->value, $config->plugin);
+                            // Get strings in nicer format for reuse
+                            $configname = $config->config;
+                            $pluginname = $config->plugin;
+                            $elementname = $pluginname.$configname;
+
+                            // Generate an admin tree
+                            $admintree = admin_get_root();
+                            $nodes = $admintree->search($config->config);
+                            $relevantobject = '';
+
+                            // Iterate through tree for specific page and elementname
+                            foreach ($nodes as $node) {
+                                if ($node->page instanceof \admin_settingpage && isset($node->page->settings->$elementname)) {
+                                    // Should only ever be reached once, so break loop
+                                    $relevantobject = $node->page->settings->$elementname;
+                                    break;
+                                }
+                            }
+
+                            // Check object is found, and for extra safety, ensure it is for the right plugin
+                            if ($relevantobject != '' && $relevantobject->plugin == $pluginname) {
```

Comment:
should the $relevantobject->plugin == $pluginname bit be inside the foreach above? Is it possible we have an array and break early before the proper one and then ignore it here?

Reply:
It shouldnt be. There is no config name collision in core, and for the plugins, the $elementname is plugin_nameconfig_name, so it will only ever break when the correct config and plugin is found. The $relevantobject->plugin check is essentially redundant, but I put it there incase my logic missed something